ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/components/text/TextComponentView.kt

```diff
@@ -57,10 +59,16 @@ internal fun TextComponentView(
         paywallState = state,
     )
 
+    // Calculate countdown only if this text is inside a countdown
+    val countdownTime = style.countdownDate?.let { date ->
+        rememberCountdownState(date).countdownTime
+    }
+
     // Process any variables in the text.
     val text = rememberProcessedText(
         state = state,
         textState = textState,
+        countdownTime = countdownTime,
```

Comment:
Hmm I wondered if we should move this to the `TextComponentState`, same as the package/tabIndex... but not sure if that could be a problem considering this will be updated every second? If there are no performance concerns, I would consider moving it there.